.github/workflows/test.yml

```diff
@@ -129,16 +130,23 @@ jobs:
             **/**.go
             go.mod
             go.sum
+            .github/workflows/test.yml
       - uses: actions/download-artifact@v3
         with:
           name: "${{ github.sha }}-${{ matrix.part }}"
         if: env.GIT_DIFF
+      - name: install libsodium
+        run: |
+          make libsodium
+        if: env.GIT_DIFF
       - name: test & coverage report creation
         env:
           USE_PRELOAD: 1,4
           SAVE_BRANCH_LAUNCH_DEPTH: 1
+          CGO_CFLAGS: -I/home/runner/work/lbm-sdk/lbm-sdk/tools/sodium/linux_amd64/include
+          CGO_LDFLAGS: -L/home/runner/work/lbm-sdk/lbm-sdk/tools/sodium/linux_amd64/lib -lsodium
```

Comment:
How about using `GITHUB_WORKSPACE` instead of hard-coded literals? It may also apply to other lines.